File: app/src/main/java/com/prashant/dronevideostream/RecordedVideosActivity.java
package com.prashant.dronevideostream;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.FileProvider;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.io.File;
import java.util.ArrayList;

public class RecordedVideosActivity extends AppCompatActivity {

    private ArrayList<File> videoFiles = new ArrayList<>();
    private ArrayList<String> fileNames = new ArrayList<>();
    private ListView listView;
    private VideoListAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_recorded_videos);


        listView = findViewById(R.id.list_view);

        // Load videos from both folders
        scanDirectory("Recordings");
        scanDirectory("DroneRecords");

        // Set custom adapter
        adapter = new VideoListAdapter(this, fileNames);
        listView.setAdapter(adapter);

        // Handle item clicks
        listView.setOnItemClickListener((adapterView, view, position, id) -> {
            File selectedFile = videoFiles.get(position);
            Uri fileUri = FileProvider.getUriForFile(this,
                    getPackageName() + ".fileprovider", selectedFile);

            Intent intent = new Intent(Intent.ACTION_VIEW);
            intent.setDataAndType(fileUri, "video/*");
            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
            startActivity(Intent.createChooser(intent, "Open with"));
        });
    }

    private void scanDirectory(String folderName) {
        File directory = new File(getExternalFilesDir(null), folderName);
        if (directory.exists()) {
            File[] files = directory.listFiles();
            if (files != null) {
                for (File f : files) {
                    if (f.getName().endsWith(".mp4") || f.getName().endsWith(".ts")) {
                        videoFiles.add(f);
                        fileNames.add(f.getName());
                    }
                }
            }
        }
    }
}

File: app/src/main/java/com/prashant/dronevideostream/RecordedVideosActivity.kt
package com.prashant.dronevideostream

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.FileProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.prashant.dronevideostream.databinding.ActivityRecordedVideosBinding
import java.io.File

class RecordedVideosActivity : AppCompatActivity() {

    private lateinit var binding: ActivityRecordedVideosBinding

    private val videoFiles = mutableListOf<File>()
    private val fileNames = mutableListOf<String>()

    private lateinit var adapter: VideoRecyclerAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityRecordedVideosBinding.inflate(layoutInflater)
        setContentView(binding.root)

        loadVideos()
        setupRecyclerView()
    }

    private fun loadVideos() {
        scanDirectory("Recordings")
        scanDirectory("DroneRecords")
    }

    private fun setupRecyclerView() {
        binding.recyclerView.layoutManager = LinearLayoutManager(this)

        val adapter = VideoRecyclerAdapter(
            videos = videoFiles,
            onClick = { file ->
                playVideo(file)
            },
            onShare = { file ->
                shareVideo(file)
            },
            onDelete = { file, position ->
                confirmDelete(file, position)
            }
        )

        binding.recyclerView.adapter = adapter
    }

    private fun confirmDelete(file: File, position: Int) {
        AlertDialog.Builder(this)
            .setTitle("Delete video?")
            .setMessage("This video will be permanently deleted.")
            .setPositiveButton("Delete") { _, _ ->
                if (file.delete()) {
                    videoFiles.removeAt(position)
                    binding.recyclerView.adapter?.notifyItemRemoved(position)
                }
            }
            .setNegativeButton("Cancel", null)
            .show()
    }

    private fun shareVideo(file: File) {
        val uri = FileProvider.getUriForFile(
            this,
            "$packageName.fileprovider",
            file
        )

        val intent = Intent(Intent.ACTION_SEND).apply {
            type = "video/*"
            putExtra(Intent.EXTRA_STREAM, uri)
            addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
        }
        startActivity(Intent.createChooser(intent, "Share video"))
    }

    private fun playVideo(file: File) {
        val uri = FileProvider.getUriForFile(
            this,
            "$packageName.fileprovider",
            file
        )

        val intent = Intent(Intent.ACTION_VIEW).apply {
            setDataAndType(uri, "video/*")
            addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
        }
        startActivity(intent)
    }


    private fun scanDirectory(folderName: String) {
        val directory = File(getExternalFilesDir(null), folderName)
        if (!directory.exists()) return

        directory.listFiles()?.forEach { file ->
            if (file.extension == "mp4" || file.extension == "ts") {
                videoFiles.add(file)
                fileNames.add(file.name)
            }
        }
    }
}
